python/ql/src/Exceptions/EmptyExcept.ql

```diff
@@ -61,27 +62,32 @@ predicate subscript(Stmt s) {
   s.(Delete).getATarget() instanceof Subscript
 }
 
-predicate encode_decode(Call ex, ClassValue type) {
+predicate encode_decode(Call ex, Expr type) {
   exists(string name | ex.getFunc().(Attribute).getName() = name |
-    name = "encode" and type = ClassValue::unicodeEncodeError()
+    name = "encode" and
+    type = API::builtin("UnicodeEncodeError").getAValueReachableFromSource().asExpr()
     or
-    name = "decode" and type = ClassValue::unicodeDecodeError()
+    name = "decode" and
+    type = API::builtin("UnicodeDecodeError").getAValueReachableFromSource().asExpr()
   )
 }
 
-predicate small_handler(ExceptStmt ex, Stmt s, ClassValue type) {
+predicate small_handler(ExceptStmt ex, Stmt s, Expr type) {
   not exists(ex.getTry().getStmt(1)) and
   s = ex.getTry().getStmt(0) and
-  ex.getType().pointsTo(type)
+  ex.getType() = type
```

Comment:
This is (at a surface level) somewhat weaker than the previous version, but I guess it might be okay?

Reply:
The logic is reshuffled a bit. In this predicate, we no longer consider a `ClassValue` but merely an expression. The relation between the expression and an appropriate class is now simpler here and more complicated in `focussed_handler` (and `encode_decode`) where the API graphs kicks in.

Reply:
Oh, good point! Thanks for clarifying. :slightly_smiling_face: